machine/v1beta1/types_azureprovider.go

```diff
@@ -227,18 +292,33 @@ type DiskSettings struct {
 	EphemeralStorageLocation string `json:"ephemeralStorageLocation,omitempty"`
 }
 
-// ManagedDiskParameters is the parameters of a managed disk.
+// ManagedDiskParameters is the parameters of a OSDisk managed disk.
 type ManagedDiskParameters struct {
-	// StorageAccountType is the storage account type to use. Possible values include "Standard_LRS" and "Premium_LRS".
+	// StorageAccountType is the storage account type to use.
+	// Possible values include "Standard_LRS", "Premium_LRS".
```

Comment:
lack of verification worries me.  *after* this PR merges, let's open a PR that fixes the validation and we'll talk about what we can and cannot do.

Reply:
Yes, we could improve this with an `kubebuilder:validation:Enum` in a follow up PR.

Reply:
Here is a draft PR where we can talk about what we can do with it: https://github.com/openshift/api/pull/1139